v1/topdown/cache/cache.go

```diff
@@ -330,62 +365,65 @@ func (c *cache) cleanStaleValues() (dropped int) {
 	return dropped
 }
 
-type InterQueryValueCache interface {
+type InterQueryValueCacheBucket interface {
 	Get(key ast.Value) (value any, found bool)
 	Insert(key ast.Value, value any) int
 	Delete(key ast.Value)
-	UpdateConfig(config *Config)
 }
 
-type interQueryValueCache struct {
-	items  map[string]any
-	config *Config
+type interQueryValueCacheBucket struct {
+	items  util.TypedHashMap[ast.Value, any]
```

Comment:
Is this more performant than the regular map?

Reply:
The cache key is a complex type that needs to be hashable/comparable; the `util.TypedHashMap` is a special case of the old `util.HashMap` that allow you to have non-`ast.Value` values.